src/orbitalMechanics.cpp

```diff
@@ -0,0 +1,405 @@
+
+            if (err < 1e-13 || numberOfIterations > 12) { //TODO magic numbers?
+              break;
+            }
+
+            Tmin = x2tof(Tmin, xNew, nMax);
+
+            xOld = xNew;
+            numberOfIterations++;
+          }
+          if (Tmin > T) {
+            nMax--;
+          }
+        }
+      }
+
+      // We exit this if clause with Mmax being the maximum number of revolutions
+      // for which there exists a solution. We crop it to m_multi_revs
+      nMax = std::min(m_multi_revs, nMax); //TODO: check m_multi_revs
+
+      // 2.2 We now allocate the memory for the output variables
+      //std::vector<arma::Col<double>::fixed<3>> v1Vec; // std::vector< std::pair< arma::Col<double>::fixed<3>, arma::Col<double>::fixed<3> > >
+      //v1Vec.resize(nMax * 2 + 1);
+      //std::vector<arma::Col<double>::fixed<3>> v2Vec;
+      //v2Vec.resize(nMax * 2 + 1);
+      std::vector<std::pair<arma::Col<double>::fixed<3>, arma::Col<double>::fixed<3>>> velocitiesVector;
+      velocitiesVector.reserve(nMax * 2 + 1);
+      //std::vector<int> itersVec;
+      //itersVec.resize(nMax * 2 + 1);
+      std::vector<double> xVector;
+      xVector.reserve(nMax * 2 + 1);
+
+      // 3 - We may now find all solutions in x,y
+      // 3.1 0 rev solution
+      // 3.1.1 initial guess
+      if (T >= T00) {
+        xVector.at(0) = -(T - T00) / (T - T00 + 4.0);
+      } else if (T <= T1) {
+        xVector.at(0) = T1 * (T1 - T) / (2.0 / 5.0 * (1.0 - lambdaSquare * lambdaCubic) * T) + 1.0;
+      } else {
+        xVector.at(0) = std::pow((T / T00), 0.69314718055994529 / std::log(T1 / T00)) - 1.0;
+      }
+
+      // 3.1.2 Householder iterations
+      //itersVec[0] = brent(...); //need??? or TODO -> householder(T, xVector[0], 0.0, 1e-5, 15);
+
+      // 3.2 multi rev solutions
+      double tmp;
+      for (int i = 1; i < nMax + 1; ++i) {
+        //3.2.1 left Householder iterations
+        tmp = std::pow((i * arma::datum::pi + arma::datum::pi) / (8.0 * T), 2.0 / 3.0);
+        xVector.at(2 * i - 1) = (tmp - 1.0) / (tmp + 1.0);
+        //itersVec[2 * i - 1] = brent(...); //need??? or TODO -> householder(T, xVector[2*i-1], i, 1e-8, 15);
+
+        //3.2.2 right Householder iterations
+        tmp = std::pow((8.0 * T) / (i * arma::datum::pi), 2.0 / 3.0);
+        xVector.at(2 * i) = (tmp - 1.0) / (tmp + 1.0);
+        //itersVec[2 * i] = brent(...); //need??? or TODO -> householder(T, xVector[2 * i], i, 1e-8, 15);
+      }
+
+      /// 4 - For each found x value we reconstruct the terminal velocities
+      double gamma = std::sqrt(muSun * lengthSumWeight / 2.0);
+      double rho = (departurePositionVectorLength - arrivalPositionVectorLength) / flightVectorLength;
+      double sigma = std::sqrt(1 - std::pow(rho, 2.0));
+
+      double vt, vr1, vt1, vr2, vt2, y;
+
+      for (size_t i = 0; i < xVector.size(); ++i) {
+        y = std::sqrt(1.0 - lambdaSquare + lambdaSquare * std::pow(xVector.at(i), 2.0));
```

Comment:
sqrt(1 - a + a * b^2) = sqrt(a * (b^2 - 1) + 1)

And use `std::pow(lambda, 2.0)` instead of `lambdaSquare`.